### PR TITLE
project description -> name with interface

### PR DESCRIPTION
**Pull request by tulinkry**

### CODE REVIEW

File: src/org/opensolaris/opengrok/configuration/Project.java [hunk 1]
```diff
@@ -204,18 +208,18 @@ public static Project getProject(File file) {
...
-     * which project is to be searched
+     * Returns project object by its name, used in webapp to figure out which
+     * project is to be searched
      *
-     * @param desc description of the project
-     * @return project that fits the description
+     * @param desc name of the project
+     * @return project that fits the name
      */
     public static Project getByDescription(String desc) {
```
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/Project.java]: getByName() ?

File: test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java [hunk 2]
```diff
@@ -235,6 +235,7 @@ public void testStoreAndGetIncrementalTags() throws Exception {
      */
     @ConditionalRun(condition = RepositoryInstalled.MercurialInstalled.class)
     @Test
+    @Ignore
```
Comment (vladak) [inline test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java]: why is this needed ?

File: test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java [hunk 3]
```diff
@@ -346,6 +347,7 @@ public void testStoreAndGet() throws Exception {
...
```
Comment (vladak) [inline test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java]: ditto

File: src/org/opensolaris/opengrok/configuration/Project.java (hunk 1 above)
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/Project.java]: changed

File: test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java (hunk 2 above)
Comment (tulinkry) [inline test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java]: my personal tuning, removed

File: test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java (hunk 3 above)
Comment (tulinkry) [inline test/org/opensolaris/opengrok/history/FileHistoryCacheTest.java]: my personal tuning, removed

**Comment by tarzanek** [comment]
lgtm, merging
